delta/sdk/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/resources/ResourcesImplSpec.scala

```diff
@@ -143,7 +161,26 @@ class ResourcesImplSpec
         forAll(list) { case (id, schemaSegment, schemaRef) =>
           val sourceWithId = source deepMerge json"""{"@id": "$id"}"""
           val expectedData = ResourceGen.resource(id, projectRef, sourceWithId, Revision(schemaRef.iri, 1))
-          val resource     = resources.create(id, projectRef, schemaSegment, sourceWithId).accepted
+          val resource     = resources.create(id, projectRef, schemaSegment, sourceWithId, None).accepted
+          resource shouldEqual ResourceGen.resourceFor(
+            expectedData,
+            types = types,
+            subject = subject
+          )
+        }
+      }
+
+      "succeed and tag with the id present on the payload and passed" in {
+        val list =
+          List(
+            (myId12, "_", resourceSchema),
+            (myId13, "myschema", Latest(schema1.id))
```

Comment:
Similar with this test: we probably need to resolve by tag afterwards

Reply:
Why's that, is there something asynchronous happening / other state mutated? The return value from `resources.create` isn't enough to check against?

Reply:
If you think about the functionality you are adding, this test doesn't assert that it has occurred. 

The outcome of creating with a tag is really that the resource is resolvable by that tag. The fact that the data returned from the `create` contains tag data does not indicate that the tag has been added correctly. The way to think about it is that I could comment out the code that persists the tag, and this test would not fail